flow/src/main/java/com/sequenceiq/flow/cleanup/InMemoryCleanup.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.flow.cleanup;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.scheduler.PollGroup;
+import com.sequenceiq.cloudbreak.cloud.store.InMemoryResourceStateStore;
+import com.sequenceiq.flow.core.FlowRegister;
+import com.sequenceiq.flow.core.chain.FlowChains;
+
+@Component
+public class InMemoryCleanup {
+
+    @Inject
+    private FlowRegister runningFlows;
+
+    @Inject
+    private FlowChains flowChains;
+
+    public void cancelEveryFlowWithoutDbUpdate() {
+        for (String resourceType : InMemoryResourceStateStore.getResourceTypes()) {
+            for (Long resourceId : InMemoryResourceStateStore.getAllResourceId(resourceType)) {
+                InMemoryResourceStateStore.putResource(resourceType, resourceId, PollGroup.CANCELLED);
+            }
+        }
+        for (String id : runningFlows.getRunningFlowIds()) {
+            String flowChainId = runningFlows.getFlowChainId(id);
+            if (flowChainId != null) {
+                flowChains.removeFullFlowChain(flowChainId);
+            }
+            runningFlows.remove(id);
```

Comment:
this is the same with `cancelFlowWithoutDbUpdate`

Reply:
true, reused.